React/src/components/EditingForm.tsx

```diff
@@ -0,0 +1,97 @@
+import Toolbar, { Item as ToolbarItem } from "devextreme-react/toolbar";
+import DataGrid, { Editing, Form, Paging } from "devextreme-react/data-grid";
+import { Item } from 'devextreme-react/form';
+import { ClickEvent } from "devextreme/ui/button";
+import notify from "devextreme/ui/notify";
+import { useCallback, useMemo, useRef } from "react";
+import service from "../data";
+
+const employees = service.getData();
+
+function EditingForm(){
+    const grid = useRef<DataGrid>(null);
+    const saveButtonClick = useCallback((e:ClickEvent) => {
```

Comment:
You can simplify this part. Remove useCallback and put this function directly in saveOptions.onClick. The same for other buttons.